tests/Feature/Http/Controllers/TradeOfferControllerTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Character;
+use App\Models\Item;
+use App\Models\Trade;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Support\Facades\Auth;
+use JMac\Testing\Traits\AdditionalAssertions;
+use Tests\TestCase;
+
+class TradeOfferControllerTest extends TestCase
+{
+    use AdditionalAssertions;
+    use RefreshDatabase;
+    use WithFaker;
+
+    public $user;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->user = User::factory()->create();
+        Auth::login($this->user);
+    }
+
+
+    /**
+     * @test
+     */
+    public function create_displays_view()
+    {
+        $trade = Trade::factory()->create();
+        $character = Character::factory()->create([
+            'user_id' => $this->user->id
+        ]);
+        $items = Item::factory()->count(2)->create([
+            'character_id' => $character->id
+        ]);
+        $character->items()->saveMany($items);
+
+        $response = $this->actingAs($this->user)->get(route('offer.create', $trade));
+
+        $response->assertOk();
+        $response->assertViewIs('offer.create');
+        $response->assertViewHas('validItems');
+        $response->assertViewHas('trade');
+    }
+
+    /**
+     * @test
+     */
+    public function store_attaches_offer_to_trade()
+    {
+        $item = Item::factory()->create([
+            'rarity' => 'rare',
+            'id' => 234,
+            'description' => "listed"
+        ]);
+
+        $trade = Trade::factory()->create([
+            'id' => 456,
+            'item_id' => $item->id
+        ]);
```

Comment:
you don't need to / shouldn't explicitly set the `id` of the model here, there's a change for a collision. When the factory returns from `create()` the id will have been generated for you.
```suggestion
            'rarity' => 'rare',
            'description' => "listed"
        ]);

        $trade = Trade::factory()->create([
            'item_id' => $item->id
        ]);
```

Reply:
This was done for debugging purposes to make it easier when using dd() to spot if the relationships contained the intended objects. But I agree, I should probably remove it after confirmation.